mipengine/tests/node/test_sql_injection_protection.py

```diff
@@ -0,0 +1,112 @@
+import json
+
+import pymonetdb
+import pytest
+
+from mipengine.common.node_tasks_DTOs import ColumnInfo, TableSchema
+from mipengine.tests.node import nodes_communication
+
+local_node = nodes_communication.get_celery_app("local_node_1")
+local_node_create_table = nodes_communication.get_celery_create_table_signature(local_node)
+local_node_get_tables = nodes_communication.get_celery_get_tables_signature(local_node)
+local_node_get_table_schema = nodes_communication.get_celery_get_table_schema_signature(local_node)
+local_node_get_table_data = nodes_communication.get_celery_get_table_data_signature(local_node)
+local_node_create_merge_table = nodes_communication.get_celery_create_merge_table_signature(local_node)
+local_node_get_merge_tables = nodes_communication.get_celery_get_merge_tables_signature(local_node)
+local_node_create_view = nodes_communication.get_celery_create_view_signature(local_node)
+local_node_get_views = nodes_communication.get_celery_get_views_signature(local_node)
+local_node_create_remote_table = nodes_communication.get_celery_create_remote_table_signature(local_node)
+local_node_get_remote_tables = nodes_communication.get_celery_get_remote_tables_signature(local_node)
+local_node_cleanup = nodes_communication.get_celery_cleanup_signature(local_node)
+
+context_id = "HISTOGRAMS"
+
+
+@pytest.fixture(autouse=True)
+def cleanup_tables():
+    yield
+
+    local_node_cleanup.delay(context_id=context_id.lower()).get()
+
+
+def test_sql_injection_get_tables():
+    with pytest.raises(ValueError):
+        local_node_get_tables.delay(context_id=");\"").get()
+
+
+def test_sql_injection_get_table_schema():
+    with pytest.raises(ValueError):
+        local_node_get_table_schema.delay(table_name="drop table data;").get()
```

Comment:
If you test for SQL injections write an actual SQL injection. I suggest [`Robert'); DROP TABLE data; --`](https://xkcd.com/327/)